src/tensor_ops/cuda_kernels.rs

```diff
@@ -146,7 +146,7 @@ impl<K: BinaryOpCudaKernel + AsKernelParam> BinaryKernel<K, f32> for Cuda {
         grad_out: &Self::Storage<S, f32>,
     ) -> Result<(), Self::Err> {
         let bwd_fn = self.dev.get_func(K::MODULE_NAME, K::BWD_FN_NAME).unwrap();
-        let numel = lhs.shape.num_elements();
+        let numel = grad_out.shape.num_elements();
```

Comment:
This shouldn't be necessary since they all have the same shape. What was happening before this was changed?

Reply:
Yeah, my bad. I think I mistook that as being the number of elements physically stored in lhs, rather than the number of virtual elements. I'll revert that change.